nautobot_chatops/dispatchers/base.py

```diff
@@ -37,6 +38,38 @@ def __init__(self, context=None):
         """Init this Dispatcher with the provided dict of contextual information (which will vary by app)."""
         self.context = context or {}
 
+    def _get_cache_key(self) -> str:
+        """Key generator for the cache, adding the plugin prefix name."""
+        # Using __file__ as a key customization within the cache
+        key_string = "-".join([__file__, self.context.get("user_name")])
+        return hashlib.md5(key_string.encode("utf-8")).hexdigest()  # nosec
+
+    @property
+    def session(self) -> dict:
```

Comment:
Instead of a single `session` dictionary, with APIs to get/set/update the dict as a whole, did you consider exposing it more directly as a key-value store? Something like:

```python
def get_session(key) -> value:
def set_session(key, value):
def unset_session(key):
```

It *feels* to me like that would be more generally useful, but you have the more direct experience with how this would be actually used, so what do you think?

Reply:
it's an extra feature that, as you pointed out, would simplify its usage!

Reply:
@glennmatthews , check this implementation https://github.com/nautobot/nautobot-plugin-chatops/pull/88/commits/c46b4cf694dbca0c7fe11a99461a4f71aedb4955
I took your idea and extended/complicated, supporting both modes, depending if a `key` is provided. I can see it useful to set/get the session data in one shoot, but I also see that this makes the methods a bit more obscure to the user.
It's kind of a tradeoff...

Reply:
@glennmatthews just a heads up for this PR, to move it forward